### PR TITLE
feat(generator): add polling policy sample

### PR DESCRIPTION
**Pull request by alevenberg**
https://github.com/googleapis/google-cloud-cpp/issues/13455

https://github.com/googleapis/google-cloud-cpp/blob/c784b75be83fdd328309993c55ca857dd3ca7524/google/cloud/spanner/samples/samples.cc#L4012-L4037

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13797)

### CODE REVIEW

**Comment by alevenberg** [comment]
I'm waiting to generate the rest of the files, until I get a review. I based it off the retry policy and took from the spanner sample that set a polling operation

To generate the second commit, I used: GENERATE_GOLDEN_ONLY=1 ci/cloudbuild/build.sh -t generate-libraries-pr

**Comment by dbolduc** [comment]
> I'm waiting to generate the rest of the files, until I get a review.

That is not necessary. We don't tend to look at the generated code anyway.

FWIW, it LGTM so far.

---

Part of the problem is to include the new samples in the docs. e.g. this is what we do for retry policies:

https://github.com/googleapis/google-cloud-cpp/blob/393e0f39d087377806c9efcc0b4179ce1d1934be/google/cloud/iap/doc/override-retry-policies.dox#L73

**Comment by coryan** [comment]
> Part of the problem is to include the new samples in the docs. e.g. this is what we do for retry policies:
> 
> https://github.com/googleapis/google-cloud-cpp/blob/393e0f39d087377806c9efcc0b4179ce1d1934be/google/cloud/iap/doc/override-retry-policies.dox#L73

Most of the work is in the scripts to update markdown:

https://github.com/googleapis/google-cloud-cpp/blob/393e0f39d087377806c9efcc0b4179ce1d1934be/ci/generate-markdown/update-library-landing-dox.sh#L131-L150

**Comment by alevenberg** [comment]
> > I'm waiting to generate the rest of the files, until I get a review.
> 
> That is not necessary. We don't tend to look at the generated code anyway.
> 
> FWIW, it LGTM so far.
> 
> Part of the problem is to include the new samples in the docs. e.g. this is what we do for retry policies:
> 
> https://github.com/googleapis/google-cloud-cpp/blob/393e0f39d087377806c9efcc0b4179ce1d1934be/google/cloud/iap/doc/override-retry-policies.dox#L73

ack i am planning on doing that in a separate PR

File: generator/internal/sample_generator.cc [hunk 1]
```diff
@@ -177,6 +177,69 @@ void SetRetryPolicy(std::vector<std::string> const& argv) {
...
+  auto options = google::cloud::Options{}
+  .set<google::cloud::$product_namespace$::$service_name$PollingPolicyOption>(
+    google::cloud::GenericPollingPolicy<google::cloud::$product_namespace$::$service_name$LimitedTimeRetryPolicy,google::cloud::ExponentialBackoffPolicy>(
+     google::cloud::$product_namespace$::$service_name$LimitedTimeRetryPolicy(
+          /*maximum_duration=*/std::chrono::minutes(45)),
+      google::cloud::ExponentialBackoffPolicy(
+          /*initial_delay=*/std::chrono::seconds(10),
+          /*maximum_delay=*/std::chrono::minutes(2),
+          /*scaling=*/4.0))
+      .clone());
```
Comment (coryan) [inline generator/internal/sample_generator.cc:203]: Consider:

```suggestion
  auto options = google::cloud::Options{}
  .set<google::cloud::$product_namespace$::$service_name$PollingPolicyOption>(
    google::cloud::GenericPollingPolicy<
        google::cloud::$product_namespace$::$service_name$RetryPolicy,
        google::cloud::BackoffPolicy>(
     google::cloud::$product_namespace$::$service_name$LimitedTimeRetryPolicy(
          /*maximum_duration=*/std::chrono::minutes(45)),
      google::cloud::ExponentialBackoffPolicy(
          /*initial_delay=*/std::chrono::seconds(10),
          /*maximum_delay=*/std::chrono::minutes(2),
          /*scaling=*/4.0))
      .clone());
```

Or maybe introduce some type aliases?

**Review by dbolduc — APPROVED** [review]

**Pull request merged**